src/Modals.tsx

```diff
@@ -59,7 +59,7 @@ export const Modals = () => {
     if (userPubkey && notifySignatureRequired) {
       if (isWeb3ModalOpen) {
         // Close web3modal in case user is switching accounts
-        closeWeb3Modal()
+        // closeWeb3Modal()
```

Comment:
### Why we required to disable this

When signing with email, we handle the signature message in the w3m, not in the wallet popup (like MM, or other wallets). It happens inside an iframe that renders [Magic](https://magic.link/)'s message approves screen. So, auto closing causes a UI regression and immediately closes the signature modal. 

Since we don't have a chance to check if the used method is email or not, we need to disable auto closing w3m. This condition is quite a rare case and if some reason the modal opens, users could close it and continue proceeding.